peer_manager.go

```diff
@@ -39,7 +43,10 @@ func (pm *peerManager) addPeer(addr ma.Multiaddr) error {
 			logger.Error(err)
 			return err
 		}
-		pm.importAddresses(resolvedAddrs)
+		pm.importAddresses(resolvedAddrs, connect)
+	}
+	if connect {
+		pm.host.Network().DialPeer(pm.ctx, pid)
```

Comment:
Stupid question: what is the purpose of Dialing the peer?

Reply:
Not so stupid, I'm not sure anymore.. I run into some problems with peers rejecting connection on certain ports when everyone bootstraps at the same time and played with the scenario of actively dialing peers first (setting up connection) rather than waiting to be contacted.

In the end I'm not sure it makes a difference. Most of my problems went away when put a lock around the PeerAdd so that not two peers can be joining in the same moment.